src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -630,6 +630,14 @@ bool SerialHeap::requires_barriers(stackChunkOop obj) const {
 
 // Returns "TRUE" iff "p" points into the committed areas of the heap.
 bool SerialHeap::is_in(const void* p) const {
+  // precondition
+  DEBUG_ONLY(verify_not_in_native_if_java_thread();)
```

Comment:
No need for `DEBUG_ONLY` now that you've added `NOT_DEBUG_RETURN`
```suggestion
  verify_not_in_native_if_java_thread();
```